packages/apm/src/hubextensions.ts

```diff
@@ -55,15 +56,17 @@ function startSpan(spanOrSpanContext?: Span | SpanContext, forceNoChild: boolean
     span = new Span(spanOrSpanContext, that);
   }
 
-  if (span.sampled === undefined && span.transaction !== undefined) {
+  // We only roll the dice on sampling for "root" spans (transactions) because the childs inherit this state
+  if (span.sampled === undefined && !span.isChildSpan()) {
     const sampleRate = (client && client.getOptions().tracesSampleRate) || 0;
```

Comment:
The `|| 0` here hides a configuration error or the unavailability of `client`, and has bitten us.

`tracesSampleRate` is already defaulted when setting the client options, I think we should not do the `|| 0` here, wdyt?

Reply:
Similarly, isn't it a programming error if `client` is `undefined`?

Shouldn't we be able to write simply:

```suggestion
    const sampleRate = client.getOptions().tracesSampleRate;
```

Reply:
`Client` can be optional on the hub, that's why we need it.

Reply:
If there is no `client`, then `sampleRate` can only be 0, okay.
Wouldn't this be a case to log a debug message? We debug log when the noop transport is in use in the Go SDK => when one sees that in the debug log, it is obvious why events are not sent. Here it seems like a similar situation, if there is no client we can't ever send any event, the SDK is somewhat disabled.